src/Signer.php

```diff
@@ -5,11 +5,13 @@
 
 use Lcobucci\JWT\Signer\CannotSignPayload;
 use Lcobucci\JWT\Signer\Ecdsa\ConversionFailed;
-use Lcobucci\JWT\Signer\InvalidKeyProvided;
-use Lcobucci\JWT\Signer\Key;
 
 interface Signer
 {
+    public static function generateKey(): mixed;
+
+    public function validateAndConsumeKey(string $contents, string $passphrase): void;
```

Comment:
1. Every signer must implement Key validation. Even `None` signer has to have it: explicit code is always better
1. I despise `validateAndConsumeKey` method name, but Signer might need to consume key before validation, see https://github.com/lcobucci/jwt/blob/ae3aac816d4e716e2cd26c5085559409a8432d46/src/Signer/OpenSSL.php#L58-L59

Reply:
Having the method doesn't guarantee its execution, I don't think we're making the code better by enforcing validation this way.